### PR TITLE
fix: Autoclose issue

### PR DESCRIPTION
**Pull request by Dhruwang**
## What does this PR do?

Fixes [1879](https://linear.app/formbricks/issue/FOR-1879/bug-auto-close-preview-is-not-working)
Also, current autoclose does not work outside preview as well (progressbar get stuck)

https://github.com/formbricks/formbricks/assets/67850763/92d08a80-12c9-4c27-a063-edb933217db0

Also refactored the progress logic, initially we were doing some calculations to derive progress value, changed it to use simple css animation

After fix:

https://github.com/formbricks/formbricks/assets/67850763/8c1bd0f1-e1d6-43f5-8152-a519aea36cec

## How should this be tested?

Test preview with close on inactivity turned on

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand bits
- [x] Ran `pnpm build`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues

### Appreciated

- [x] If a UI change was made: Added a screen recording or screenshots to this PR
- [x] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

File: packages/surveys/src/styles/global.css [hunk 1]
```diff
@@ -82,3 +82,12 @@ p.fb-editor-paragraph {
   --fb-close-btn-color: var(--slate-500);
   --fb-close-btn-color-hover: var(--slate-700);
 }
+
+@keyframes ShrinkWidthToZero {
```
Comment (mattinannt) [inline packages/surveys/src/styles/global.css:86]: As we currently don't have a naming convention for keyframes I would either prefer lower camel case from javascript `shrinkWidthToZero` or `shrink-width-to-zero` as the other css classes.

**Review by mattinannt — COMMENTED** [review]
@Dhruwang I'm very happy to see that you even made improvements to the naming in this PR so the codebase follows the code conventions 😊💪
There are two things I would still like to see improved:

File: packages/surveys/src/components/wrappers/AutoCloseWrapper.tsx [hunk 2]
```diff
@@ -1,59 +1,53 @@
...
+    return stopCountdown;
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [survey.autoClose]);
 
   return (
-    <>
-      {!countdownStop && survey.autoClose && <Progress progress={countdownProgress} />}
-      <div onClick={handleStopCountdown} onMouseOver={handleStopCountdown} className="h-full w-full">
+    <div>
+      {survey.autoClose && showAutoCloseProgressBar && <AutoCloseProgressBar timeout={survey.autoClose} />}
```
[the remark this answers is not in the document]
Comment (Dhruwang) [reply packages/surveys/src/components/wrappers/AutoCloseWrapper.tsx:47]: Changed it to `autoCloseTimeout` 😊

File: packages/surveys/src/styles/global.css (hunk 1 above)
Comment (Dhruwang) [reply packages/surveys/src/styles/global.css:86]: Changed it to `shrink-width-to-zero` , so that it aligns with other css classes as well 😊

**Review by mattinannt — APPROVED** [review]
@Dhruwang thanks for making the changes :-) Looks good! 😊

**Pull request merged**